app/uk/gov/hmrc/fileupload/controllers/EnvelopeController.scala

```diff
@@ -50,21 +48,24 @@ class EnvelopeController(withBasicAuth: BasicAuth,
 
   def create() = Action.async(jsonBodyParser[CreateEnvelopeRequest]) { implicit request =>
     def envelopeLocation = (id: EnvelopeId) => LOCATION -> s"${ request.host }${ uk.gov.hmrc.fileupload.controllers.routes.EnvelopeController.show(id) }"
-    val command = CreateEnvelope(nextId(), request.body.callbackUrl, request.body.expiryDate, request.body.metadata)
+    val command = CreateEnvelope(nextId(), request.body.callbackUrl, request.body.expiryDate, request.body.metadata, CreateEnvelopeRequest.constraintsWithDefaultsIfNotProvided(request.body.constraints, envelopeDefaultConstraints))
```

Comment:
This `constraintsWithDefaultsIfNotProvided` does not seem right to me. You should be only passing `request.body.constraints` here and defaults should come into play if needed later in the code (no need to pass them separately each time there is a create envelope request.)